src/hyperion/parameters/gridscan.py

```diff
@@ -0,0 +1,263 @@
+from __future__ import annotations
+
+from typing import Any
+
+import numpy as np
+from dodal.devices.detector import DetectorParams
+from dodal.devices.fast_grid_scan import GridScanParams
+from dodal.devices.zebra import (
+    RotationDirection,
+)
+from pydantic import validator
+from scanspec.core import Path as ScanPath
+from scanspec.specs import Line
+
+from hyperion.external_interaction.ispyb.ispyb_dataclass import (
+    GridscanIspybParams,
+)
+from hyperion.parameters.components import (
+    DiffractionExperiment,
+    OptionalGonioAngleStarts,
+    RotationAxis,
+    TemporaryIspybExtras,
+    WithSample,
+    WithScan,
+    XyzAxis,
+    XyzStarts,
+)
+from hyperion.parameters.constants import CONST
+
+
+class GridCommon(DiffractionExperiment, OptionalGonioAngleStarts, WithSample):
+    grid_width_um = CONST.PARAM.GRIDSCAN.WIDTH_UM
+    exposure_time_s: float = CONST.PARAM.GRIDSCAN.EXPOSURE_TIME_S
+    use_roi_mode: bool = CONST.PARAM.GRIDSCAN.USE_ROI
+    transmission_frac: float = 1
+    # field rather than inherited to make it easier to track when it can be removed:
+    ispyb_extras: TemporaryIspybExtras = TemporaryIspybExtras()
+
+
+class GridScanWithEdgeDetect(GridCommon, WithSample): ...
+
+
+class PinTipCentreThenXrayCentre(GridCommon):
+    tip_offset_um: float = 0
+
+
+class RobotLoadThenCentre(GridCommon, WithSample): ...
+
+
+class SpecifiedGridScan(GridCommon, XyzStarts, WithScan, WithSample):
```

Comment:
Should: Can you add some docstrings around what specified means in this context?